job-server/test/spark.jobserver/LocalContextSupervisorSpec.scala

```diff
@@ -80,8 +82,8 @@ class LocalContextSupervisorSpec extends TestKit(LocalContextSupervisorSpec.syst
 
     it("should be able to add multiple new contexts") {
       supervisor ! AddContext("c1", contextConfig)
-      supervisor ! AddContext("c2", contextConfig)
       expectMsg(ContextInitialized)
+      supervisor ! AddContext("c2", contextConfig)
```

Comment:
Yes we need to make localConextSupervisor a singleton.  It is that right now, need to think of a good way to enforce that.   Don't worry about Await, it's not the right way since its blocking. 

-Evan
"Never doubt that a small group of thoughtful, committed citizens can change the world" - M. Mead

> On Jan 18, 2015, at 5:02 AM, Fernando Otero <notifications@github.com> wrote:
> 
> In job-server/test/spark.jobserver/LocalContextSupervisorSpec.scala:
> 
> >        expectMsg(ContextInitialized)
> > +      supervisor ! AddContext("c2", contextConfig)
> Answering to the 2 previous emails in one :D 1- yes, we are not stoping the context, but that only means we were able to instantiate it, so there's not guarantee of what will happen if you do that, so creating multiple contexts should be left at user's discretion and marked as risky :D 2- I'm adding a commit with just the comments in the PR, but so just to confirm: making LocalContextSupervisorActor#startContext a blocking action (using Await) seems to solve the issue, But what I understand of akka (which is little) it seems like one can create several actor instances, so even making that call synchronous will not solve the issue if another LocalContextSupervisorActor gets spawned and calls startContext on its own,  We need a way to guarantee that there will only be one instance of whoever creates the SparkContext() What I'm saying is: I can put all the AddContext requests in a queue but then I need a guarantee that that queue will have only one listener, which is basically a singleton actor (for which I couldn't find documentation other than Cluster Singleton <http://doc.akka.io/docs/akka/snapshot/contrib/cluster-singleton.html>.)
> …
> —
> Reply to this email directly or view it on GitHub.
>